src/Metrics.tcl

```diff
@@ -254,5 +254,250 @@ proc report_design_area_metrics {args} {
+
+    set inp_to_reg_critical_path [lindex [find_timing_paths -path_delay max -from [all_inputs] -to [all_registers]] 0]
+    if {$inp_to_reg_critical_path != ""} {
+      set target_clock_latency_max [sta::format_time [$inp_to_reg_critical_path target_clk_delay] 4]
+    } else {
+      set target_clock_latency_max 0
+    }
+
+
+    set inp_to_reg_critical_path [lindex [find_timing_paths -path_delay min -from [all_inputs] -to [all_registers]] 0]
+    if {$inp_to_reg_critical_path != ""} {
+      set target_clock_latency_min [sta::format_time [$inp_to_reg_critical_path target_clk_delay] 4]
+      set source_clock_latency [sta::format_time [$inp_to_reg_critical_path source_clk_latency] 4]
+    } else {
+      set target_clock_latency_min 0
+      set source_clock_latency 0
+    }
+
+    report_puts "\n=========================================================================="
+    report_puts "$when critical path target clock latency max path"
+    report_puts "--------------------------------------------------------------------------"
+    report_puts "$target_clock_latency_max"
+
+    report_puts "\n=========================================================================="
+    report_puts "$when critical path target clock latency min path"
+    report_puts "--------------------------------------------------------------------------"
+    report_puts "$target_clock_latency_min"
+
+    report_puts "\n=========================================================================="
+    report_puts "$when critical path source clock latency min path"
+    report_puts "--------------------------------------------------------------------------"
+    report_puts "$source_clock_latency"
+    } else {
+    puts "No registers in design"
+    }
+    # end if all_registers
+
+    report_puts "\n=========================================================================="
+    report_puts "$when critical path delay"
+    report_puts "--------------------------------------------------------------------------"
+    report_puts "$path_delay"
+
+    report_puts "\n=========================================================================="
+    report_puts "$when critical path slack"
+    report_puts "--------------------------------------------------------------------------"
+    report_puts "$path_slack"
+
+    report_puts "\n=========================================================================="
+    report_puts "$when slack div critical path delay"
+    report_puts "--------------------------------------------------------------------------"
+    report_puts "[format "%4f" [expr $path_slack / $path_delay * 100]]"
+  }
+
+  report_puts "\n=========================================================================="
+  report_puts "$when report_power"
+  report_puts "--------------------------------------------------------------------------"
+  if {[info exists ::env(CORNERS)]} {
```

Comment:
This should use the corners defined in sta

Reply:
So we need to check sta::env(CORNERS) ?

Reply:
you need to use `sta::corners` which returns the list of defined corners.

Reply:
Thanks for noticing. I changed the code.